core/relay/adaptor/text-embeddings-inference/adaptor.go

```diff
@@ -0,0 +1,140 @@
+	}
+}
+
+// text-embeddings-inference api see https://huggingface.github.io/text-embeddings-inference/#/Text%20Embeddings%20Inference/rerank
+
+func (a *Adaptor) SetupRequestHeader(meta *meta.Meta, _ *gin.Context, req *http.Request) error {
+	switch meta.Mode {
+	case mode.Rerank:
+		req.Header.Set("Content-Type", "application/json")
+	case mode.Embeddings:
+		req.Header.Set("Content-Type", "application/json")
+	default:
+		return fmt.Errorf("unsupported mode: %s", meta.Mode)
+	}
+	return nil
+}
+
+func (a *Adaptor) ConvertRequest(meta *meta.Meta, req *http.Request) (string, http.Header, io.Reader, error) {
+	switch meta.Mode {
+	case mode.Rerank:
+		// Parse request body into AST node
+		node, err := common.UnmarshalBody2Node(req)
+		if err != nil {
+			return "", nil, nil, fmt.Errorf("failed to parse request body: %w", err)
+		}
+
+		// Get the documents array and rename it to texts
+		documentsNode := node.Get("documents")
+		if !documentsNode.Exists() {
+			return "", nil, nil, errors.New("documents field not found")
+		}
+
+		// Get the raw documents value
+		documentsValue, err := documentsNode.Raw()
+		if err != nil {
+			return "", nil, nil, fmt.Errorf("failed to get documents value: %w", err)
+		}
+
+		// Set the texts field with the documents value
+		_, err = node.Set("texts", ast.NewString(documentsValue))
+		if err != nil {
+			return "", nil, nil, fmt.Errorf("failed to set texts field: %w", err)
+		}
+
+		// Remove the documents field
+		_, err = node.Unset("documents")
+		if err != nil {
+			return "", nil, nil, fmt.Errorf("failed to remove documents field: %w", err)
+		}
+
+		// Convert back to JSON
+		jsonData, err := node.MarshalJSON()
+		if err != nil {
+			return "", nil, nil, fmt.Errorf("failed to marshal request: %w", err)
+		}
+
+		return "", nil, bytes.NewReader(jsonData), nil
+	case mode.Embeddings:
+		// Handle embeddings request
+		return "", nil, req.Body, nil
+	default:
+		return "", nil, nil, fmt.Errorf("unsupported mode: %s", meta.Mode)
+	}
+}
+
+func (a *Adaptor) DoRequest(_ *meta.Meta, _ *gin.Context, req *http.Request) (*http.Response, error) {
+	return utils.DoRequest(req)
+}
+
+func (a *Adaptor) DoResponse(_ *meta.Meta, c *gin.Context, resp *http.Response) (*model.Usage, *relaymodel.ErrorWithStatusCode) {
+	// Read response body
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, openai.ErrorWrapper(err, "read_response_body_failed", http.StatusInternalServerError)
+	}
+
+	// Check for error status
+	if resp.StatusCode != http.StatusOK {
+		return nil, openai.ErrorWrapperWithMessage(
+			fmt.Sprintf("request failed with status %d", resp.StatusCode),
+			"upstream_error",
+			resp.StatusCode,
+		)
+	}
+
+	// Write response back to client
+	c.Header("Content-Type", "application/json")
+	c.Status(resp.StatusCode)
+	if _, err := c.Writer.Write(body); err != nil {
+		return nil, openai.ErrorWrapper(err, "write_response_failed", http.StatusInternalServerError)
+	}
```

Comment:
The format of the returned data is not consistent, refer to the defined structure:
```go
type RerankResponse struct {
	Meta    RerankMeta      `json:"meta"`
	ID      string          `json:"id"`
	Results []*RerankResult `json:"results"`
}
```